src/components/com_tjcertificate/administrator/libraries/certificate.php

```diff
@@ -788,8 +814,19 @@ public function issueCertificate($replacements, $options)
 				throw new Exception(Text::_('COM_TJCERTIFICATE_TEMPLATE_INVALID'));
 			}
 
-			// Generate unique certificate id
-			$this->unique_certificate_id = $this->generateUniqueCertId($options);
+			$issuedCertInfo = $this::getIssued($this->client, $this->client_id, $this->user_id);
```

Comment:
Don't do this. Just check if `$this->unique_certificate_id` is already set, if not then continue with `$this->unique_certificate_id = $this->generateUniqueCertId($options);`

Reply:
You will need to set `$this->unique_certificate_id` in `addCertEntry` while certificate regeneration

Reply:
done